app/javascript/components/upload/WizardNavPanel.js

```diff
@@ -50,5 +57,75 @@ function RawWizardNavPanel({
+  }
+  const metadataFiles = serverState.files.filter(metadataFileFilter)
+  const isMetadataParsing = metadataFiles.some(f => f.parse_status === 'parsing')
+  const isMetadataParsed = metadataFiles.some(f => f.parse_status === 'parsed')
+
+  const clusteringFiles = serverState.files.filter(clusterFileFilter)
+  const isClusteringParsing = isMetadataParsing || clusteringFiles.some(f => f.parse_status === 'parsing')
+  const isClusteringParsed = isMetadataParsed && clusteringFiles.some(f => f.parse_status === 'parsed')
+
+  const processedExpFiles = serverState.files.filter(processedFileFilter)
+  const isExpressionParsing = isMetadataParsing || processedExpFiles.some(f => f.parse_status === 'parsing')
+  const isExpressionParsed = isMetadataParsed && processedExpFiles.some(f => f.parse_status === 'parsed')
+
+  let clusterStatusMsg = <span className="success">clustering visuals <FontAwesomeIcon icon={faCheck}/></span>
+  if (!isClusteringParsed) {
+    clusterStatusMsg = <span className="warning">no cluster visuals <FontAwesomeIcon icon={faQuestionCircle}/></span>
+  }
+  const clusterVizIndicator = <div>
+    <OverlayTrigger
+      trigger={['hover', 'focus']}
+      rootClose placement="top"
+      overlay={clusteringHelpContent(isClusteringParsing)}>
+      <span>
+        {clusterStatusMsg} { isClusteringParsing && <LoadingSpinner/> }
+      </span>
+    </OverlayTrigger>
+  </div>
+
+  let expressionStatusMsg = <span className="success">expression visuals <FontAwesomeIcon icon={faCheck}/></span>
+  if (!isExpressionParsed) {
+    expressionStatusMsg = <span className="warning">no expression visuals <FontAwesomeIcon icon={faQuestionCircle}/></span>
+  }
+  const expressionVizIndicator = <div>
+    <OverlayTrigger
+      trigger={['hover', 'focus']}
+      rootClose placement="top"
+      overlay={expressionHelpContent(isExpressionParsing)}>
+      <span>
+        {expressionStatusMsg} { isExpressionParsing && <LoadingSpinner/> }
+      </span>
+    </OverlayTrigger>
+  </div>
+  return <div className="viz-notifications">
+    {clusterVizIndicator}
+    {expressionVizIndicator}
+  </div>
+}
+
+/** gets the popup message based on whether there are files parsing */
+function expressionHelpContent(isExpressionParsing) {
+  return <Popover id="expression-viz-upload-info" className="tooltip-wide">
+    <div> A processed matrix file and a metadata file are required for gene expression visualization </div>
+    { isExpressionParsing && parsingMessage }
+  </Popover>
+}
+
+/** gets the popup message based on whether there are files parsing */
+function clusteringHelpContent(isClusteringParsing) {
+  return <Popover id="cluster-viz-upload-info" className="tooltip-wide">
+    <div> A metadata file and a clustering file are required for cluster visualization</div>
+    { isClusteringParsing && parsingMessage }
+  </Popover>
+}
+
+const parsingMessage = <div> Some files which will impact visualization are currently being processed </div>
```

Comment:
Let's end sentences with a period.  (The other changes here are a bit more subjective.)

```suggestion
const parsingMessage = <div>Some files that will impact visualization are being processed.</div>
```